runtime/doc/lua.txt

```diff
@@ -2990,6 +2990,33 @@ vim.fs.parents({start})                                     *vim.fs.parents()*
         (string|nil)
 
 
+==============================================================================
+Lua module: vim.glob                                                *vim.glob*
+
+vim.glob.to_lpeg({pattern})                               *vim.glob.to_lpeg()*
+    Parses a raw glob into an |lua-lpeg| pattern.
+
+    This uses glob semantics from LSP 3.17.0: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#pattern
```

Comment:
For reference, the "glob semantics" is the main (current) reason in favor of a `vim.glob` module as opposed to "in situ" functions like `vim.lpeg.to_glob()`, `vim.regex.to_glob()`, etc: the glob semantics need to live somewhere.

OTOH, we could have a private `vim._glob` function which centralizes the semantics for use by in-situ functions.